### PR TITLE
add logging

### PR DESCRIPTION
**Pull request by titan-dn**
- print UEK kernels list for migrating from Oracle
- print stdout to /var/log/almalinux-deploy.log

### CODE REVIEW

File: almalinux-deploy.sh [hunk 1]
```diff
@@ -360,6 +362,16 @@ grub_update() {
     fi
 }
 
+# Check do we have custom kernel (e.g. kernel-uek) and print warning
+check_custom_kernel() {
+output=$(rpm -qa | grep kernel-uek) || :
```
Comment (soksanichenko) [inline almalinux-deploy.sh:367]: var `output` should be local

File: almalinux-deploy.sh [hunk 2]
```diff
@@ -360,6 +362,16 @@ grub_update() {
     fi
 }
 
+# Check do we have custom kernel (e.g. kernel-uek) and print warning
+check_custom_kernel() {
```
Comment (soksanichenko) [inline almalinux-deploy.sh:366]: use indent for body of function

File: almalinux-deploy.sh [hunk 3]
```diff
@@ -360,6 +362,16 @@ grub_update() {
...
+output=$(rpm -qa | grep kernel-uek) || :
+if [ -n "${output}" ]; then
```
Comment (soksanichenko) [inline almalinux-deploy.sh:368]: Why do we not use double brackets?

File: almalinux-deploy.sh [hunk 4]
```diff
@@ -360,6 +362,16 @@ grub_update() {
...
+    echo -e "[31;1mYou've had kernels that now can't be booting[0m:"
+    # shellcheck disable=SC2001,SC2086
+    echo $output | sed 's# #\n#'
```
Comment (soksanichenko) [inline almalinux-deploy.sh:371]: better `echo "${output}"` than this

File: almalinux-deploy.sh [hunk 5]
```diff
@@ -360,6 +362,16 @@ grub_update() {
     fi
 }
 
+# Check do we have custom kernel (e.g. kernel-uek) and print warning
+check_custom_kernel() {
+output=$(rpm -qa | grep kernel-uek) || :
+if [ -n "${output}" ]; then
+    echo -e "[31;1mYou've had kernels that now can't be booting[0m:"
```
Comment (soksanichenko) [inline almalinux-deploy.sh:369]: I think it's not informative message for user. I suggest to add info about what should do a client in this case and why its server can't be booting

**Review by soksanichenko — CHANGES_REQUESTED** [review]

File: almalinux-deploy.sh (hunk 3 above)
Comment (titan-dn) [reply almalinux-deploy.sh:368]: it doesn't matter in this case

**Review by soksanichenko — APPROVED** [review]

File: almalinux-deploy.sh [hunk 6]
```diff
@@ -360,6 +362,21 @@ grub_update() {
     fi
 }
 
+# Check do we have custom kernel (e.g. kernel-uek) and print warning
+check_custom_kernel() {
+    local output
+    output=$(rpm -qa | grep kernel-uek) || :
+    if [ -n "${output}" ]; then
+        echo -ne "\n!! [31;1mYou've had kernels that can't be booted in
```
Comment (andrewlukoshko) [inline almalinux-deploy.sh:370]: There are kernels left from previous operating system:

**Review by soksanichenko — APPROVED** [review]
Looks ok

File: almalinux-deploy.sh [hunk 7]
```diff
@@ -360,6 +362,25 @@ grub_update() {
     fi
 }
 
+# Check do we have custom kernel (e.g. kernel-uek) and print warning
+check_custom_kernel() {
+    local output
+    output=$(rpm -qa | grep kernel-uek) || :
+    if [ -n "${output}" ]; then
+        if [ -x /usr/bin/mokutil ] && /usr/bin/mokutil --sb-state 2>&1 | grep -q enabled; then
+            echo -ne "\n!! [31;1mYou've had kernels that can't be booted in
```
Comment (andrewlukoshko) [inline almalinux-deploy.sh:371]: There are kernels left from previous operating system that won't boot in Secure Boot mode anymore:

**Pull request merged**